delivery_carrier_label_postlogistics/models/delivery.py

```diff
@@ -109,18 +109,20 @@ class DeliveryCarrierOption(models.Model):
 
     name = fields.Char(translate=True)
 
-    tmpl_option_id_domain = fields.Char(
-        compute='_compute_tmpl_option_id_domain'
+    allowed_tmpl_options_ids = fields.Many2many(
+        'delivery.carrier.template.option',
+        compute='_compute_allowed_tmpl_options_ids',
+        store=False,
     )
 
-    @api.depends('carrier_id.allowed_options_domain')
-    def _compute_tmpl_option_id_domain(self):
-        """ Gets the domain from related delivery.carrier
-            (be it from cache or context) """
+    @api.depends('carrier_id.allowed_tmpl_options_ids')
+    def _compute_allowed_tmpl_options_ids(self):
+        """ Gets the available template options from related delivery.carrier
+            (be it from cache or context)."""
         for option in self:
-            domain = self.env.context.get('default_tmpl_option_id_domain')
-            option.tmpl_option_id_domain = (
-                option.carrier_id.allowed_options_domain or domain or '[]')
+            domain = self.env.context.get('default_allowed_tmpl_options_ids')
+            option.allowed_tmpl_options_ids = (
+                option.carrier_id.allowed_tmpl_options_ids or domain or '[]')
```

Comment:
`or '[]'` seems quite dangerous to use here, since you're writing to `Many2many` :smile: